apps/passport-server/src/services/semaphoreService.ts

```diff
@@ -169,13 +177,33 @@ export class SemaphoreService {
 
   private async reloadZuzaluGroups(): Promise<void> {
     return traced("Semaphore", "reloadZuzaluGroups", async (span) => {
-      const users = await fetchAllLoggedInZuzaluUsers(this.dbPool);
+      const zuzaluUsers: LoggedInZuzaluOrZuconnectUser[] =
+        await fetchAllLoggedInZuzaluUsers(this.dbPool);
+      const zuconnectUsers = await fetchAllLoggedInZuconnectUsers(this.dbPool);
+      // Give Zuconnect users roles equivalent to Zuzalu roles
+      const zuconnectUsersWithZuzaluRoles = zuconnectUsers.map((user) => {
+        return {
+          email: user.attendee_email,
+          role: zuconnectProductIdToZuzaluRole(user.product_id),
+          commitment: user.commitment
+        };
+      });
+      // If the same user appears with the same role in both Zuzalu and
+      // Zuconnect, only use one of them
+      const users = _.uniqWith(
+        zuzaluUsers.concat(zuconnectUsersWithZuzaluRoles),
+        (a, b) =>
+          a.role === b.role &&
+          a.commitment === b.commitment &&
+          a.email === b.email
+      );
```

Comment:
I think it's important to use the most powerful role - i.e. if they have a bunch of visitor roles and one resident role, use the resident role. if they have a bunch of visitor resident and organizer tickets, use the organizer role. etc.

Reply:
Yes, this only eliminates duplicates in cases where the roles are equal. If the roles are different, both go through and the user ends up in all of the relevant groups. Possibly this bit is redundant given a later piece of code, so I'll look at removing it.